code/game/objects/items/devices/flashlight.dm

```diff
@@ -237,6 +231,46 @@
 		attack_self(usr)
 		return TRUE
 
+/obj/item/device/flashlight/lamp/proc/toggle_onoff(var/onoff = null) //this is only called by gameticker.dm at roundstart, so we call update_brightness() with playsound = FALSE below.
+	if(isnull(onoff))
+		on = !on
+	else
+		on = onoff
+	update_brightness(playsound = FALSE)
+
+//Lamps draw power from the area they're in, unlike flashlights.
+/obj/item/device/flashlight/lamp/New()
+	if(drawspower)
+		pwrconn = new(src)
```

Comment:
did you check to see if you move the lamp to a new area that it appropriately deducts power from the new area?

Reply:
It does turn off properly when it's taken out into space, but actually I'm in the process of rewriting a few things. I'll test it a bit more after that. 
It seems like lamps being turned "on" at roundstart in occupied departments was broken in the last few months, so I'm trying to puzzle that out currently. I might make them start out "off" again since if #33449 passes it will seemingly make lights being off and then turning on after a few seconds a non-issue anyway?